app/controllers/posts_controller.rb

```diff
@@ -535,16 +535,16 @@ def toggle_comments
   end
 
   def lock
-    return not_found unless current_user.privilege? 'flag_curate'
+    return not_found unless current_user&.privilege? 'flag_curate'
     return not_found if @post.locked?
 
     length = params[:length].present? ? params[:length].to_i : nil
     if length
-      if !current_user.is_moderator && length > 30
+      if !current_user&.at_least_moderator? && length > 30
```

Comment:
Untested suggestion:

```suggestion
      if current_user&.standard? && length > 30
```

Reply:
Same as with other inversions of `at_least_moderator?` + safe navigation operator + negation - not as simple as inverting to `standard?`